### PR TITLE
Fix various errors reported by goreportcard

### PR DESCRIPTION
**Pull request by antham**

### CODE REVIEW

**Comment by antham** [comment]
I hope it was not introduced with golang CI enabled. There is still an ineffassign error https://goreportcard.com/report/github.com/jesseduffield/lazygit#ineffassign, I see there is a panic earlier in the code to handle error, if I'm right you planned to get rid of those panic which are all around the code ?

**Comment by antham** [comment]
Codecov is not happy maybe because there is no coverage yet here

File: pkg/commands/os.go [hunk 1]
```diff
@@ -175,7 +175,8 @@ func (c *OSCommand) Unquote(message string) string {
 	return message
 }
 
-func (C *OSCommand) AppendLineToFile(filename, line string) error {
+// AppendLineToFile adds a new line at the beginning of a given string
```
Comment (jesseduffield) [inline pkg/commands/os.go]: It actually adds the given string as a new line to the given file :)

**Comment by jesseduffield** [comment]
@antham yep I plan on removing all the panics

**Comment by glvr182** [comment]
@jesseduffield @antham let's wait with documenting things for after the transition, otherwise we would get too many conflicts between merges perhaps

**Comment by antham** [comment]
@glvr182 Which transition ?

**Comment by glvr182** [comment]
@antham we're going to restructure the repo to a gitflow repo

**Comment by antham** [comment]
ah ok right

**Comment by glvr182** [comment]
@antham i could fill you in on the details on slack if you want

**Comment by remyabel** [comment]
The ineffectual assignment here:
```
	err = bIter.ForEach(func(b *plumbing.Reference) error {
		name := b.Name().Short()
		branches = append(branches, commands.Branch{Name: name})
		return nil
	})

return branches
```
is just because `err` is not actually being used.

**Comment by antham** [comment]
@remyabel yep it's the purpose of this test, the problem is that we don't handle gracefully error in this function we panic for the err which is above, so if we handle error properly there it would make sense to remove panic from the whole file in the same time.

**Comment by antham** [comment]
@glvr182 I don't know, I've read the issue but if you think it could be useful, I'm not against the idea.

**Comment by remyabel** [comment]
I'm for documenting ASAP to reduce technical debt.

**Comment by remyabel** [comment]
No, I was referring to glvr182's comment.

**Comment by antham** [comment]
Ah ok sorry

**Comment by glvr182** [comment]
@remyabel Yea documenting new code is good but what i mean is since there multiple things going on right now, be careful to not cause merge conflicts.

**Comment by antham** [comment]
@glvr182 it's not a problem to have merge conflict, it's part of collaboration, especially if you rebase regularly it's not a big deal.

File: pkg/commands/os.go (hunk 1 above)
Comment (antham) [reply pkg/commands/os.go]: yep that's stupid

**Comment by antham** [comment]
@jesseduffield if you can come back

**Comment by jesseduffield** [comment]
All looks good to me, mergin'

**Pull request merged**